tla-types/src/test/scala/at/forsyte/apalache/tla/typecheck/etc/TestEtcTypeChecker.scala

```diff
@@ -521,7 +521,7 @@ class TestEtcTypeChecker extends AnyFunSuite with EasyMockSugar with BeforeAndAf
 
   test("annotated polymorphic let-definition") {
     // The names of type variables may change during type inference
-    // @type: a => b;
+    // @type: a => a;
     // let F == lambda x \in Set(a): ((c, c) => c) x d in F(Int)
```

Comment:
This expression is written in the syntax of EtcExpr, a lambda calculus over types. The body of the lambda is an application of `x` and `d` to a type, which in this case is an operator of two arguments of type `c` and returns type `c`.

Reply:
It is an identity from the types of view.

Reply:
Thank you for explaining. It helps a lot. So is the lambda here a type-level lambda? It seems like it's kind of mixing type level syntax with pseudo-TNT syntax, or am I just totally missing the thread here :D?

> It is an identity from the types of view.

So, we are maybe getting outside the scope of this PR,  but one thing that I think is throwing me off here is that, afaik, it is not possible to write an operator that actually has type `a => a` and also relies on some constant `d` that comes from outside the operator.

The only way we can write an operator of type `a => a` is if we treat the provided argument as fully abstract, because we can't know anything about the value of type `a`, since we don't know what type it is (or, equivalently, it must work for any type). So afaik, there would be no way of having a `d` available that could work for any possible type for application to `(c, c) => c`. afaiu, This is why (in a HM-ish type system) [any function with type `a => a` must be (isomorphic to) the identity function](https://stackoverflow.com/questions/12230820/haskell-example-of-function-of-type-a-a-besides-the-identity). 

The fact that this test is going through makes me think that either our type inference is off, or we're constructing type here that shouldn't actually be possible to assign to any expression (or else I am just quite confused about what is going on here). But in any of these three cases, it seems like we might be looking to either fix the test, fix the type inference, or fix my understanding :)